### PR TITLE
Hide teleport.internal labels from UI

### PR DESCRIPTION
**Pull request by sielakos**
Solves the [issue of hiding teleport.internal](https://github.com/gravitational/webapps/issues/1081) labels. 

As mentioned in the previous PR it should be hidden only in UI. 
That could be achieved in 2 ways:
1. Hidding it on the browser's side with JS code
2. Hidding it on the REST endpoint side (which makes gRPC calls to auth service)

After analysis of the FE code I come to two I had 2 observations:
1. It would require changing a few places in the code base responsible for fetching each resource type. 
2. It would be slower because we need sometimes to make multiple requests to display one page of the table.

That's why I took the second approach because it enables me to:
1. Change only one place in code bases used by all resource-related endpoints
2. REST API server should be in the same cluster as Auth Service, so while fetching one page can require multiple requests, the roundtrip cost should be much lower than from FE. 

The approach not requiring a round trip requires [too much interference](https://github.com/gravitational/teleport/pull/18428#discussion_r1021698292) in the teleport core.

### CODE REVIEW

**Comment by sielakos** [comment]
It's not yet completely ready, for review, I just wanted CI checks to run.

File: lib/web/resources.go [hunk 1]
```diff
@@ -351,7 +407,99 @@ func listResources(clt resourcesAPIGetter, r *http.Request, resourceKind string)
...
+		return fetchMoreResponse{Response: response}, err
+	}
+
+	hasMore := len(response.Resources) > 0
+	response = removeTeleportInternalResources(response)
+
+	return fetchMoreResponse{Response: response, HasMore: hasMore}, err
+}
+
+func removeTeleportInternalResources(response *types.ListResourcesResponse) *types.ListResourcesResponse {
```
Comment (zmb3) [inline lib/web/resources.go:472]: I think you're still misunderstanding the requirements here. 

The issue to to hide the _labels_ themselves, not hide resources that have internal labels.

**Comment by sielakos** [comment]
Sorry, I was very much basing my assumption on description in original
issue.



czw., 17 lis 2022, 15:34 użytkownik Zac Bergquist ***@***.***>
napisał:

**Comment by sielakos** [comment]
As it was pointed out I wrongly understood what the ticket is about. So closing this PR.

**Pull request closed without merging**